_source/_posts/2020-11-23-quick-graphql-react-app.md

```diff
@@ -0,0 +1,576 @@
+        <SecureRoute path='/ExchangeRates' component={ExchangeRates} />
+        <Route path='/login' render={() => <Login baseUrl={baseDomain} issuer={issuer} />} />
+        <Route path='/implicit/callback' component={LoginCallback} />
+    </Security>
+  );
+};
+
+export default AppWithRouterAccess;
+```
+
+The primary job of this file is to define the routes in your application.  This file is also where you will set up your Okta client and integrate it into your application.  You will notice the `ExchangeRates` route is defined as `SecureRoute`.  This way, the application will enforce authentication on that route.  You also need to define a route for `/implicit/callback` to ensure that Okta's login redirect URI has a place to land.  
+
+Next, you will need to update your `App.js` page with some custom code to replace the boilerplate from create-react-app.
+
+```javascript
+import React from 'react';
+import { BrowserRouter as Router } from 'react-router-dom';
+import AppWithRouterAccess from './AppWithRouterAccess';
+import 'bootstrap/dist/css/bootstrap.min.css';
+
+const App = () => {
+  return (
+    <Router>
+      <AppWithRouterAccess/>
+    </Router>
+  );
+}
+
+export default App;
+```
+
+This code will tell your application to use your new `AppWithRouterAccess` page.  You are also importing bootstrap here, as react-bootstrap requires the CSS to be imported.  This only needs to be imported once.  
+
+### Header Component
+
+Add a new folder to the `src` directory called `Components`.  Add a file called `Header.jsx` to this folder and add the following code:
+
+```javascript
+import React from 'react';
+import { useOktaAuth } from '@okta/okta-react';
+import { Navbar, Nav, Form, Button } from 'react-bootstrap'
+
+const Header = () => {
+  const { authState, authService } = useOktaAuth();
+
+  if (authState.isPending) {
+    return <div>Loading...</div>;
+  }
+
+  const button = authState.isAuthenticated ?
+    <Button variant="secondary" onClick={() => { authService.logout() }}>Logout</Button> :
+    <Button variant="secondary" onClick={() => { authService.login() }}>Login</Button>
+
+  return (
+    <Navbar bg="light" expand="lg">
+      <Navbar.Brand href="/">Lets Escape</Navbar.Brand>
+      <Navbar.Toggle aria-controls="basic-navbar-nav" />
+      <Navbar.Collapse id="basic-navbar-nav">
+        <Nav className="mr-auto">
+          <Nav.Link href="/">Home</Nav.Link>
+          <Nav.Link href="/ExchangeRates">Exchange Currency</Nav.Link>
+        </Nav>
+        <Form inline>
+          {button}
+        </Form>
+      </Navbar.Collapse>
+    </Navbar>
+  );
+};
+
+export default Header;
+```
+
+The header will be used on every page, which is why you create it as a component.  The component itself is fairly simple.  There are a couple navigation links to the pages you will implement shortly.  There is also a button that will change between Login or Logout, depending on the authentication status of the user.  
+
+### Home Page
```

Comment:
This could probably use a more SEO-friendly title, like `Create a Home Page with React`

Reply:
Third level headers don't matter for SEO